catalog/dags/elasticsearch_cluster/healthcheck_dag.py

```diff
@@ -0,0 +1,158 @@
+"""
+Monitor staging and production Elasticsearch cluster health endpoint.
+
+Requests the cluster health and alerts under the following conditions:
+
+- Red cluster health
+- Unexpected number of nodes
+- Unresponsive cluster
+
+Additionally, the DAG will notify (rather than alert) when the cluster health is yellow.
+Yellow cluster health may or may not be an issue, depending on whether it is expected,
+and occurs whenever shards and replicas are being relocated (e.g., during reindexes).
+It is worthwhile to notify in these cases, as an assurance, but we could choose to add
+logic that ignores yellow cluster health during data refresh or other similar operations.
+"""
+
+import json
+import logging
+from datetime import datetime
+
+from airflow.decorators import dag, task
+from airflow.providers.elasticsearch.hooks.elasticsearch import ElasticsearchPythonHook
+from elasticsearch import Elasticsearch
+
+from common.constants import PRODUCTION, STAGING, Environment
+from common.slack import send_alert, send_message
+from elasticsearch_cluster.shared import get_es_host
+
+
+logger = logging.getLogger(__name__)
+
+
+_DAG_ID = "{env}_elasticsearch_cluster_healthcheck"
+
+EXPECTED_NODE_COUNT = 6
+EXPECTED_DATA_NODE_COUNT = 3
+EXPECTED_MASTER_NODE_COUNT = 3
+
+
+def _format_response_body(response_body: dict) -> str:
+    return f"""
+    Full healthcheck response body:
+    ```
+    {json.dumps(response_body, indent=4)}
+    ```
+    """
```

Comment:
The indentation on this message appears a little odd with the inserted response body:

````
[2024-02-07, 21:29:55 UTC] {slack.py:321} INFO - 
    Elasticsearch staging cluster node count is **1**.
    Expected 6 total nodes.
    Master nodes: **0** of expected 3
    Data nodes: **1** of expected 3
    This is a critical status change, **investigate ASAP**.
    If this is expected (e.g., during controlled node or cluster changes), acknowledge immediately with explanation.
    
    Full healthcheck response body:
    ```
    {
    "cluster_name": "docker-cluster",
    "status": "yellow",
    "timed_out": false,
    "number_of_nodes": 1,
    "number_of_data_nodes": 1,
    "active_primary_shards": 38,
    "active_shards": 38,
    "relocating_shards": 0,
    "initializing_shards": 0,
    "unassigned_shards": 38,
    "delayed_unassigned_shards": 0,
    "number_of_pending_tasks": 0,
    "number_of_in_flight_fetch": 0,
    "task_max_waiting_in_queue_millis": 0,
    "active_shards_percent_as_number": 50.0
}
    ```
````

Do you mind adding `textwrap.dedent` around these so they're left aligned? It might require a little fiddling with the `json.dumps` string since the closing bracket for that appears to be completely unindented.

Reply:
Sure. It ended up being janky in the end in code, but it's just a consequence of interpolating a multi-line string that has its own indentation built in. But now the messages look right: https://github.com/WordPress/openverse/pull/3748/commits/53d8f8e069452c78bf59edce119a7b6c8b02e624